Automattic/wp-calypso
# Checkout: Refactor to use react-query for cached contact details

#### Proposed Changes

Currently cached contact details in checkout come from an API endpoint and are stored in Redux state. They are then queried by a React hook inside checkout (which also fetches them if they are not loaded) and that data is provided to the checkout forms. This involves a lot of moving parts because of Redux boilerplate (actions, selectors, reducer, etc.). Due to persistence it can also cause the data to be out-of-date fairly easily.

In this PR we refactor the hook used by checkout so that it uses the `react-query` library to manage its data instead of Redux.

To do:

- [ ] Force update of cache on checkout load (without `refetchOnMount` since that seems to break tests).
- [ ] Add QueryClient and mocked contact details to unrelated checkout tests.
- [ ] Replace cached contact details updates with `react-query` also.
- [ ] Replace other uses of cached contact details with the new hook.
- [ ] Remove cached contact details from Redux.

#### Testing Instructions

- Add a product to your cart and visit checkout.
- Fill out the checkout form; if there are already cached contact details, change them to something else so you'll be able to tell the difference.
- Reload the page.
- Verify that the cached contact details show the changed fields.
